cmake/OpenCVModule.cmake

```diff
@@ -629,7 +629,7 @@ macro(ocv_create_module)
   if(OPENCV_MODULE_${the_module}_HEADERS AND ";${OPENCV_MODULES_PUBLIC};" MATCHES ";${the_module};")
     foreach(hdr ${OPENCV_MODULE_${the_module}_HEADERS})
       string(REGEX REPLACE "^.*opencv2/" "opencv2/" hdr2 "${hdr}")
-      if(hdr2 MATCHES "^(opencv2/.*)/[^/]+.h(..)?$")
+      if(hdr2 MATCHES "^(opencv2/?.*)/[^/]+.h(..)?$")
```

Comment:
I'd suggest to use more "obvious" regexp here:
```.cmake
if(hdr2 MATCHES "^opencv2/([^/]+/)*[^/]+\\.(h|hpp)$")
```
Your variant will match paths like _"opencv2test/something.hpp"_ and I think main idea is to allow only headers from _"opencv2/"_ folders. 

However it seems to be non-critical because all headers are located in "<module>/include/opencv2" folders.